detector.go

```diff
@@ -242,14 +281,31 @@ func (bp *BuildpackTOML) detect(c *DetectConfig) detectRun {
 	if err := cmd.Run(); err != nil {
 		if err, ok := err.(*exec.ExitError); ok {
 			if status, ok := err.Sys().(syscall.WaitStatus); ok {
-				return detectRun{Code: status.ExitStatus(), Output: out.Bytes()}
+				return DetectRun{Code: status.ExitStatus(), Output: out.Bytes()}
 			}
 		}
-		return detectRun{Code: -1, Err: err, Output: out.Bytes()}
+		return DetectRun{Code: -1, Err: err, Output: out.Bytes()}
 	}
-	var t detectRun
+	var t DetectRun
 	if _, err := toml.DecodeFile(planPath, &t); err != nil {
-		return detectRun{Code: -1, Err: err}
+		return DetectRun{Code: -1, Err: err}
+	}
+	if api.MustParse(bp.API).Equal(api.MustParse("0.2")) {
+		if t.hasInconsistentVersions() || t.Or.hasInconsistentVersions() {
+			t.Err = errors.Errorf(errInconsistentVersion, bp.Buildpack.Name)
+			t.Code = -1
+		}
+	}
+	if api.MustParse(bp.API).Compare(api.MustParse("0.3")) >= 0 {
+		if t.hasDoublySpecifiedVersions() || t.Or.hasDoublySpecifiedVersions() {
+			t.Err = errors.Errorf(errDoublySpecifiedVersions, bp.Buildpack.Name)
```

Comment:
```suggestion
			t.Err = errors.Errorf(errInconsistentVersion, bp.Buildpack.ID)
```

`ID` is required but `Name` is optional